applications/GeoMechanicsApplication/custom_elements/U_Pw_base_element.cpp

```diff
@@ -28,64 +29,38 @@ int UPwBaseElement::Check(const ProcessInfo& rCurrentProcessInfo) const
     // Base class checks for positive area and Id > 0
     if (int ierr = Element::Check(rCurrentProcessInfo); ierr != 0) return ierr;
 
-    const PropertiesType& rProp = this->GetProperties();
-    const GeometryType&   rGeom = this->GetGeometry();
+    const PropertiesType& r_properties = this->GetProperties();
+    const GeometryType&   r_geometry = this->GetGeometry();
 
-    // verify nodal variables and dofs
-    for (unsigned int i = 0; i < this->GetGeometry().PointsNumber(); ++i) {
-        if (!rGeom[i].SolutionStepsDataHas(DISPLACEMENT))
-            KRATOS_ERROR << "missing variable DISPLACEMENT on node " << rGeom[i].Id() << std::endl;
-
-        if (!rGeom[i].SolutionStepsDataHas(VELOCITY))
-            KRATOS_ERROR << "missing variable VELOCITY on node " << rGeom[i].Id() << std::endl;
-
-        if (!rGeom[i].SolutionStepsDataHas(ACCELERATION))
-            KRATOS_ERROR << "missing variable ACCELERATION on node " << rGeom[i].Id() << std::endl;
-
-        if (!rGeom[i].SolutionStepsDataHas(WATER_PRESSURE))
-            KRATOS_ERROR << "missing variable WATER_PRESSURE on node " << rGeom[i].Id() << std::endl;
-
-        if (!rGeom[i].SolutionStepsDataHas(DT_WATER_PRESSURE))
-            KRATOS_ERROR << "missing variable DT_WATER_PRESSURE on node " << rGeom[i].Id() << std::endl;
-
-        if (!rGeom[i].SolutionStepsDataHas(VOLUME_ACCELERATION))
-            KRATOS_ERROR << "missing variable VOLUME_ACCELERATION on node " << rGeom[i].Id() << std::endl;
-
-        if (!rGeom[i].HasDofFor(DISPLACEMENT_X) || !rGeom[i].HasDofFor(DISPLACEMENT_Y) ||
-            (this->GetGeometry().WorkingSpaceDimension() > 2 && !rGeom[i].HasDofFor(DISPLACEMENT_Z)))
-            KRATOS_ERROR << "missing one of the dofs for the variable DISPLACEMENT on node "
-                         << rGeom[i].Id() << std::endl;
-
-        if (!rGeom[i].HasDofFor(WATER_PRESSURE))
-            KRATOS_ERROR << "missing the dof for the variable WATER_PRESSURE on node "
-                         << rGeom[i].Id() << std::endl;
-    }
+    CheckUtilities::CheckNodalVariables(r_geometry, {DISPLACEMENT, VELOCITY, ACCELERATION, WATER_PRESSURE,
+                                                DT_WATER_PRESSURE, VOLUME_ACCELERATION});
+    CheckUtilities::CheckNodalDof(r_geometry, {DISPLACEMENT_X, DISPLACEMENT_Y, WATER_PRESSURE});
```

Comment:
If you'd follow my suggestion of using `Geo::ConstVariableDataRefs`, then these calls need to be modified a bit:
```suggestion
    CheckUtilities::CheckNodalVariables(r_geometry, {std::cref(DISPLACEMENT), std::cref(VELOCITY), std::cref(ACCELERATION), std::cref(WATER_PRESSURE),
                                                std::cref(DT_WATER_PRESSURE), std::cref(VOLUME_ACCELERATION)});
    CheckUtilities::CheckNodalDof(r_geometry, {std::cref(DISPLACEMENT_X), std::cref(DISPLACEMENT_Y), std::cref(WATER_PRESSURE)});
```

And I'd like to say that your utility function makes the code so much easier to read and understand. Thank you.

Reply:
done